jax/experimental/mosaic/gpu/core.py

```diff
@@ -639,6 +646,7 @@ def as_gpu_kernel(
     kernel_name: str | None = None,
     ir_version: int | None = None,
     thread_semantics: LoweringSemantics = LoweringSemantics.Lane,
+    input_output_aliases: tuple[tuple[int, int], ...] = (),
```

Comment:
The addition of input_output_aliases is a separate change. Please send an independent PR for this